g3/git/git.py

```diff
@@ -0,0 +1,21 @@
+from pydantic import BaseModel
+
+from g3.git.shell import Shell
+
+
+class GitInfo(BaseModel):
+    repo_owner: str
+    repo: str
+    branch: str
+
+
+def get_git_info() -> GitInfo:
+    sh = Shell()
+    assert sh.is_git()
+
+    branch = sh.git("rev-parse", "--abbrev-ref", "HEAD")
+    repo_info = sh.git("config", "--get", "remote.origin.url").split("/")
+    repo = repo_info[-1]
+    repo_owner = repo_info[-2]
```

Comment:
How about encapsulating this functionality in the `GitInfo` class's initializater?

```
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.__post_init__()

    def __post_init__(self):
        self.branch = sh.git("rev-parse", "--abbrev-ref", "HEAD")

        repo_info = sh.git("config", "--get", "remote.origin.url").split("/")
        self.repo = repo_info[-1]
        self.repo_owner = repo_info[-2]
```